### PR TITLE
ansible-test pylint blacklist plugin: correctly identify modules and module_utils paths for collections

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
As request by @mattclay in #65219.

I hope I picked the correct paths :)

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ansible-test

### CODE REVIEW

File: test/lib/ansible_test/_data/sanity/pylint/plugins/blacklist.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def is_module_path(path):
     :type path: str
     :rtype: bool
     """
-    return '/lib/ansible/modules/' in path or '/lib/ansible/module_utils/' in path
+    return '/lib/ansible/modules/' in path or '/plugins/modules/' in path or '/module_utils/' in path
```
Comment (mattclay) [inline test/lib/ansible_test/_data/sanity/pylint/plugins/blacklist.py]:
```suggestion
    return '/lib/ansible/modules/' in path or '/lib/ansible/module_utils/' in path or '/plugins/modules/' in path or '/plugins/module_utils/' in path
```

File: changelogs/fragments/ansible-test-blacklist-module-paths.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- "ansible-test - the pylint sanity test now correctly identifies ``modules`` and ``module_utils`` paths for collections."
```
Comment (mattclay) [inline changelogs/fragments/ansible-test-blacklist-module-paths.yml]:
```suggestion
- "ansible-test - the pylint blacklist sanity test now correctly identifies ``modules`` and ``module_utils`` paths for collections."
```

**Comment by felixfontein** [comment]
Should I add these to ignore.txt?
```
test/integration/targets/collections/collection_root_user/ansible_collections/testns/testcoll/plugins/modules/uses_base_mu_granular_nested_import.py:15:4: ansible-bad-function: Call exit_json or fail_json instead of sys.exit
test/integration/targets/collections/collection_root_user/ansible_collections/testns/testcoll/plugins/modules/uses_leaf_mu_flat_import.py:15:4: ansible-bad-function: Call exit_json or fail_json instead of sys.exit
test/integration/targets/collections/collection_root_user/ansible_collections/testns/testcoll/plugins/modules/uses_leaf_mu_granular_import.py:15:4: ansible-bad-function: Call exit_json or fail_json instead of sys.exit
test/integration/targets/collections/collection_root_user/ansible_collections/testns/testcoll/plugins/modules/uses_leaf_mu_module_import_from.py:21:4: ansible-bad-function: Call exit_json or fail_json instead of sys.exit
```

**Comment by mattclay** [comment]
@felixfontein Thanks for opening this. Those failures were a good reminder that the previous implementation I had was a hack anyways, so I've decided it's time for a proper fix. I've opened https://github.com/ansible/ansible/pull/65526 to address the issue and will close this PR.

**Pull request closed without merging**